client-js/app/elements/labrad-plot.ts

```diff
@@ -562,11 +578,54 @@ export class Plot extends polymer.Base {
               .attr('y', (d) => p.yScale(p.yNext[d[1]]))
               .attr('width', (d) => p.xScale(p.xNext[d[0]]) - p.xScale(d[0]))
               .attr('height', (d) => Math.abs(p.yScale(p.yNext[d[1]]) - p.yScale(d[1])))
-              .style('fill', (d) => getColor(d[2], zMin, zMax));
+              .style('fill', (d) => getColor(d[p.displayTraces[0] + 2], zMin, zMax));
       }
       break;
     }
   }
+  
+  private selectTraces() {
+    this.$.traceSelector.open();
+  }
+
+  private selectAll() {
+    var checkboxes: any = document.getElementsByName('traces');
+    for (var i = 0; i < checkboxes.length; i++){
+      checkboxes[i].checked = true;
+    }
+  }
+
+  private deselectAll() {
+    var checkboxes: any = document.getElementsByName('traces');
+    for (var i = 0; i < checkboxes.length; i++){
+      checkboxes[i].checked = false;
+    }
+  }
+
+  private submitTraces() {
+    var selected: Array<string> = [];
+    var serialized: Array<string>;
+    switch (this.numIndeps) {
+      case 1:
+        serialized = this.$.selectForm.serialize().traces;
```

Comment:
Instead of serializing the form, I would suggest getting the checkbox list as in the `selectAll` method above and then reading the `checked` property.